From f49051bd4aa212e5a27be5d7247683f3f024dadc Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Sun, 19 Dec 2021 10:02:19 -0500
Subject: [PATCH 1/2] Remove shebangs from non-script sources
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These files have no “if __name__ == "__main__"” or similar and no
interesting side effects. They are only meant to be imported. Therefore,
a shebang line (#!) is not useful.
---
 doc/conf.py              | 1 -
 hybridizer/hybrid.py     | 1 -
 hybridizer/io.py         | 1 -
 hybridizer/probes.py     | 1 -
 hybridizer/spikes.py     | 1 -
 hybridizer/threads.py    | 1 -
 hybridizer/validation.py | 1 -
 7 files changed, 7 deletions(-)

diff --git a/doc/conf.py b/doc/conf.py
index 30c1bca..af7e051 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # SHYBRID documentation build configuration file, created by
diff --git a/hybridizer/hybrid.py b/hybridizer/hybrid.py
index fe7ffc7..16acd11 100644
--- a/hybridizer/hybrid.py
+++ b/hybridizer/hybrid.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 SHYBRID
diff --git a/hybridizer/io.py b/hybridizer/io.py
index 7c07acd..0f56279 100644
--- a/hybridizer/io.py
+++ b/hybridizer/io.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 SHYBRID
diff --git a/hybridizer/probes.py b/hybridizer/probes.py
index f45c99a..5154d52 100644
--- a/hybridizer/probes.py
+++ b/hybridizer/probes.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 SHYBRID
diff --git a/hybridizer/spikes.py b/hybridizer/spikes.py
index 79d4166..b94920b 100644
--- a/hybridizer/spikes.py
+++ b/hybridizer/spikes.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 SHYBRID
diff --git a/hybridizer/threads.py b/hybridizer/threads.py
index 7d2f408..adc1c5b 100644
--- a/hybridizer/threads.py
+++ b/hybridizer/threads.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 SHYBRID
diff --git a/hybridizer/validation.py b/hybridizer/validation.py
index 426891a..733f3ed 100644
--- a/hybridizer/validation.py
+++ b/hybridizer/validation.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 SHYBRID

From 5aad6253969d7fc1d6cb2f36b8b8a716a5b6ee6a Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Sun, 19 Dec 2021 10:08:32 -0500
Subject: [PATCH 2/2] Remove shebangs from non-executable scripts
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These *could* use a shebang line, but the executable bit is not set, so
it will never be useful.

Leaving the shebang lines and making the files executable would be an
alternative, but:

  - For setup.py, the traditional invocation is something like “python3
    setup.py …”, and this will work fine without a shebang.

  - For hybridizer/shybrid.py, end-users will run the entry point script
    “shybrid” generated by setuptools instead, and developers can still
    run “PYTHONPATH="${PWD}" python3 hybridizer/shybrid.py” or
    “PYTHONPATH="${PWD}" python3 -m hybridizer.shybrid” if they like.
---
 hybridizer/shybrid.py | 1 -
 setup.py              | 2 --
 2 files changed, 3 deletions(-)

diff --git a/hybridizer/shybrid.py b/hybridizer/shybrid.py
index 3f213df..fa912cf 100644
--- a/hybridizer/shybrid.py
+++ b/hybridizer/shybrid.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 Created on Fri Sep 28 13:36:01 2018
diff --git a/setup.py b/setup.py
index 189849e..b39ddd4 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-
 from setuptools import setup
 
 with open("README.md", "r") as fh:
